### PR TITLE
Compile DMA domain by default for i.MX with Zephyr & enable tracing on i.MX platforms

### PR DESCRIPTION
**Pull request by dbaluta**
DMA domain with Zephyr works fine for i.MX platforms. So, remove "experimental" from description and enable it by default for i.MX platforms.

Second patch enables tracing for i.MX platforms with zephyr.

### CODE REVIEW

**Comment by paulstelian97** [comment]
Code itself looks good, I just need to be convinced it's properly tested.

**Review by iuliana-prodan — APPROVED** [review]

**Review by paulstelian97 — APPROVED** [review]
Conditional approve: compilation must not break on platforms like TGL like the previous revision. If that is true, then this looks good. (and from a cursory look that doesn't seem to be a problem anymore)

**Comment by dbaluta** [comment]
@paulstelian97  Fixed compilation problems. Also, manually tested on i.MX8MP. Will start a nightly test run to see if there are any issues reported.

File: zephyr/Kconfig [hunk 1]
```diff
@@ -29,13 +29,11 @@ config SOF_ZEPHYR_STRICT_HEADERS
 	  If unsure, say n.
 
 config DMA_DOMAIN
-	bool "Experimental: Enable the usage of DMA domain."
+	bool "Enable the usage of DMA domain."
+	default y if IMX
 	default n
```
Comment (LaurentiuM1234) [inline zephyr/Kconfig:34]: nitpick: is `default n` really required? It should be set to `n` if CONFIG_IMX is not enabled. Otherwise, LGTM.

Comment (dbaluta) [reply zephyr/Kconfig:34]: Yes, I think you are right. It is set to `n`. Will update.

**Review by paulstelian97 — APPROVED** [review]
I will accept this, though I still need to understand how the default value works in kconfig if no "default" applies.

**Review by LaurentiuM1234 — APPROVED** [review]

**Comment by dbaluta** [comment]
> I will accept this, though I still need to understand how the default value works in kconfig if no "default" applies.

When kconfig entry doesn't have a default clause it goes directly to `N`.

**Review by lgirdwood — APPROVED** [review]

**Pull request merged**